### PR TITLE
feat(package)/#92: Improve generated messages file

### PR DESCRIPTION
**Pull request by simosavonen**

### CODE REVIEW

File: package/src/cli/commands/generate/generateDistFiles.test.ts [hunk 1]
```diff
@@ -183,7 +181,12 @@ export const clientMessages = {
 	"en": {
 		"namespace.message": "Namespaced message"
 	}
-};
+} as const;
+
+export const messages = {
+	"en": { ...serverOnlyMessages["en"], ...clientMessages["en"] },
+	"fi": { ...serverOnlyMessages["fi"], ...clientMessages["fi"] }
```
Comment (Jon1VK) [inline package/src/cli/commands/generate/generateDistFiles.test.ts:188]: Order of these needs to be swapped for test to pass.

```suggestion
	"fi": { ...serverOnlyMessages["en"], ...clientMessages["fi"] },
	"en": { ...serverOnlyMessages["fi"], ...clientMessages["en"] }
```

**Comment by Jon1VK** [comment]
Great job! There was still some prettier style errors in `generateDistFiles.ts` file. Could you format the file with prettier?

**Comment by simosavonen** [comment]
Possibly missing test for edge cases. What happens if you have clientKeys regex set, but it doesn't match any messages?

**Comment by Jon1VK** [comment]
> Possibly missing test for edge cases. What happens if you have clientKeys regex set, but it doesn't match any messages?

@simosavonen In that situation the client messages object would have no messages.

```ts
export const clientMessages = {
  "fi": {},
  "en": {}
} as const;
```
It is a valid use case if none of the messages are used in the client side. The docs even mention, that:

> If you want to exclude all messages from being provided to the client, use an empty array.

Maybe a test with an empty array would be a great addition, since also the array version of the option s not tested yet. Empty array test would cover both of these cases. Would you like to add it in this PR or should I make another issue?

**Comment by simosavonen** [comment]
> Maybe a test with an empty array would be a great addition, since also the array version of the option is not tested yet. Empty array test would cover both of these cases. Would you like to add it in this PR or should I make another issue?

Added a test for `clientKeys: []`
